### PR TITLE
Implement dynamic meta tags

### LINKED ISSUE #76: ✨ [Feature request]: Change meta information based on the route
**Is your feature request related to a problem? Please describe.**
What we share a play with its deep link URL, it takes the Reactplay platform's meta(og, Twitter) image, title, and description. Rather, it should display the cover image, title, and description of the play that we are sharing on social media.

**Describe the solution you'd like**
I do not have a solution proposed yet but the contributor can start a discussion with me to brainstorm.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by erayalkis**
# Description

This Pull Request implements dynamic meta tags for all plays using react-helmet.

Fixes #76

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Inspecting the <head> section in the document inspector or copying and pasting the current url into a platform that supports embeds should let anyone test the new functionality.
On a local server though, you will have to use an external browser extension to verify that the meta tags are working as expected.
I didn't write any tests or make any changes to the documentation, as I didn't think it was necessary.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Notes:

I should mention that while I was testing the meta tags on localhost, I had to provide full paths for the meta tag (ex: "www.reactplay.io/static/exampleimage12345.png"), and I removed the domain from url before the final commit. If this causes the meta image to not work it should be super easy to fix, but I just thought I should note it down here.

I hope the new changes are good, let me know if I need to anything else !!

### CODE REVIEW

File: src/common/playlists/PlayMeta.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  let metaImage; // Initialize metaImage variable
+  if (cover) {
+    metaImage = cover; // If cover path is given, use that
+  } else {
+    try {
+      // If not, try finding the cover.png in the play's folder
+      metaImage = require(`../../plays/${playFolder}/cover.png`);
+    } catch {
+      // If no image is available, cover stays as undefined
+      console.log("No cover available.");
```
Comment (atapas) [inline src/common/playlists/PlayMeta.jsx:16]: If no image is available, let us set the top-level meta image than keep it blank? DefMeta will take care of it already?

**Review by atapas — COMMENTED** [review]
Do not commit package lock file., yarn lock enough.

File: src/common/playlists/PlayMeta.jsx (hunk 1 above)
Comment (erayalkis) [reply src/common/playlists/PlayMeta.jsx:16]: Ah, I knew I forgot something! I'll implement that ASAP, I think simply just not setting a meta tag if the metaImage variable is undefined should fix this!

Comment (erayalkis) [reply src/common/playlists/PlayMeta.jsx:16]: Done! Now, if a play doesn't have a cover image linked to it, it'll just display the default one

**Comment by atapas** [comment]
Hello @erayalkis 

We have upgraded to React v18 as part of this issue: https://github.com/atapas/react-play/issues/87

We have tested the existing plays. All good. Please pull this change to your branch and keep it up-to-date. You have to do sanity of things to make sure everything works for your changes in your branch.

Happy Coding!

**Comment by erayalkis** [comment]
I'm on it! Will ensure that everything works and push the updated version ASAP

**Comment by erayalkis** [comment]
All done, everything works as intended!

**Comment by erayalkis** [comment]
The new components introduce no UI or logic changes, so I had to test it manually using an external app, it's why the links look different on the screenshots. All of the pages look the same:
image
image
image
And when you post a link of a play on a platform that supports embeds (I personally used discord here), it looks like this:
image
If there's no cover image available, it just uses the default one instead.

**Review by atapas — APPROVED** [review]
@erayalkis Looks good.

**Comment by atapas** [comment]
@all-contributors please add @erayalkis  for Code

**Comment by erayalkis** [comment]
Will do! :)

**Pull request merged**